src/extensions/extension-discovery.ts

```diff
@@ -208,12 +208,17 @@ export class ExtensionDiscovery {
   };
 
   /**
-   * Uninstalls extension by path.
+   * Uninstalls extension.
    * The application will detect the folder unlink and remove the extension from the UI automatically.
-   * @param absolutePath Path to the non-symlinked folder of the extension
+   * @param extension Extension to unistall.
    */
-  async uninstallExtension(absolutePath: string) {
-    logger.info(`${logModule} Uninstalling ${absolutePath}`);
+  async uninstallExtension({ absolutePath, manifest }: InstalledExtension) {
+    logger.info(`${logModule} Uninstalling ${manifest.name}`);
+
+    // remove the symlink under node_modules.
+    // If we don't remove the symlink, the uninstall would leave a non-working symlink,
+    // which wouldn't be fixed if the extension was reinstalled, causing the extension not to work.
+    await fs.remove(this.getInstalledPath(manifest.name));
```

Comment:
Is there any possibility that symlink does not exist? Should we handle that case?

Reply:
.remove does nothing if the path doesn't exist so this should be safe.